Modules/sre_lib.h

```diff
@@ -500,18 +500,18 @@ do { \
     DO_JUMPX(jumpvalue, jumplabel, nextpattern, 0)
 
 typedef struct {
-    Py_ssize_t last_ctx_pos;
-    Py_ssize_t jump;
     const SRE_CHAR* ptr;
     const SRE_CODE* pattern;
     Py_ssize_t count;
-    Py_ssize_t lastmark;
-    Py_ssize_t lastindex;
+    int32_t lastmark;
+    int32_t lastindex;
     union {
         SRE_CODE chr;
         SRE_REPEAT* rep;
     } u;
-    int toplevel;
+    Py_ssize_t last_ctx_pos;
+    int8_t jump;
+    int8_t toplevel;
```

Comment:
It may save some memory, but wouldn't reading/writing a single byte be slower than reading/writing a machine word on modern computers?

Reply:
This is my reason, is it right? 
* The "cache line" of some CPUs is 64 bytes, 64 bytes are read and written at a time as a whole. So if accessing a 4-byte integer that crosses a "cache line" boundary, need to access memory twice.
* If the integer don't cross "cache line" boundary, there should be no difference between 1 byte and 4 bytes.

Reply:
I reverted `commit 9` and `commit 10`, they can be done in another issue.